File: Segundo_Curso/Segundo_Cuatri/PDOO/java/src/irrgarten/controller/Controller.java
package irrgarten.controller;

import irrgarten.Directions;
import irrgarten.Game;
import irrgarten.UI.*;


public class Controller {
    
    private Game game;
    //private TextUI view;
    private UI view;
    
    //public Controller(Game game, TextUI view) {
    public Controller(Game game, UI view) {
        this.game = game;
        this.view = view;
    }
    
    public void play() {
        boolean endOfGame = false;
        while (!endOfGame) {

            view.showGame(game.getGameState()); 
            Directions direction = view.nextMove(); 
            endOfGame = game.nextStep(direction);
        }

        view.showGame(game.getGameState());        
    }
        
}

File: Segundo_Curso/Segundo_Cuatri/PDOO/ruby/Irrgarten/Controller/controller.rb
module Control

  class Controller
    def initialize(game,view)
      @game = game
      @view = view
    end

    def play
      end_of_game = false
      while (!end_of_game)
        @view.show_game(@game.getGameState)
        direction = @view.next_move
        end_of_game = @game.nextStep(direction)
      end
      @view.show_game(@game.getGameState)
    end
  end # class
end # module
